Bash code cleanup and improved SIGTERM behaviour

cstim opened the pull request:
The bash code for vernemq docker startup was very difficult to read. This PR cleans up some of the readability issues, and fixes behaviour on SIGTERM as well.

Behaviour changes:
- After sending the SIGTERM to the process PID, we must wait for the completion of this PID, and also exit with code 143 instead of code 0 (otherwise the sigterm handler gets called many times quickly on shutdown, which looks super-ugly in the logs)
- In the SIGTERM handler, only lookup kubernetes stuff if the existing files indicate that we are indeed on kubernetes (and not on non-k8s Docker)
- All curl commands now use the option `-sS` which does not print the progress bar anymore but would still print errors

Cleanups:
- Use variables for all filenames, instead of repeating the file paths many times throughout the file
- Use variables for the common parts of the curl URLs, instead of repeating the URLs many times throughout the file
- Use variable for Authentication header
- Add some more space between redirect operators and filenames

ioolkos commented:
@cstim wow, thanks :) I'll try to review this as quickly as possible.

ioolkos commented:
> * Note: Is the call `vmq-admin node stop` also needed in a non-cluster mode? The current change would not run `vmq-admin node stop` (line 200) if the kubernetes-related directory does not exist, hence skipping this step completely, which is a behaviour change compared to before.

@cstim I think to just stop the node cleanly (or attempt to). What was your thinking in not including it? would it not be necessary for a Docker termination?

cstim commented:
> > * Note: Is the call `vmq-admin node stop` also needed in a non-cluster mode? The current change would not run `vmq-admin node stop` (line 200) if the kubernetes-related directory does not exist
> 
> @cstim I think to just stop the node cleanly (or attempt to). What was your thinking in not including it? would it not be necessary for a Docker termination?

Added again for the non-k8s to keep the behaviour from before.

> > In line 202/203 in the SIGTERM handler the NAMESPACE variable gets its value from the file, but in the rest of vernemq.sh it uses the value as defined in line 71

> Hm, I don't think it needs to be a lookup, but tbh I'm not sure there was a potential use case for this. Be safe and keep it like that?

Kept as it is (with a comment added).

Rebased, to make sure the recent commits are used as well. Remarks are considered. Can this hopefully be merged at some point?
